### PR TITLE
Added pagination and sorting to /get_all_offers

### PR DESCRIPTION
**Pull request by Quexington**
This would be a breaking DB change because I had to place new importance on the enum values behind the statuses and therefore I had to reorder them.  I'm open to suggestions on better methods of doing the relevance sorting.

### CODE REVIEW

**Comment by emlowe** [comment]
If we expect the number of such offers to be reasonably small, we could consider sorting all in python?
Alternatively, you could add a map that represents the sort mapping without changing the existing enum values.

File: chia/wallet/trading/trade_store.py [hunk 1]
```diff
@@ -246,6 +247,30 @@ async def get_all_trades(self) -> List[TradeRecord]:
 
         return records
 
+    async def get_trades_between(
+        self, start: int, end: int, sort_key: Optional[str] = None, reverse: bool = False
```
Comment (paninaro) [inline chia/wallet/trading/trade_store.py:251]: `reverse` isn't being used

File: chia/wallet/trading/trade_store.py [hunk 2]
```diff
@@ -246,6 +247,30 @@ async def get_all_trades(self) -> List[TradeRecord]:
...
+        Return a list of trades sorted by a key and between a start and end index.
+        """
+        limit = end - start
+
+        records = await self.get_all_trades()
+        if sort_key is None or sort_key == "CONFIRMED_AT_HEIGHT":
+            records = sorted(records, key=attrgetter('confirmed_at_index'), reverse=True)
+        elif sort_key == "RELEVANCE":
+            sorted_records = sorted(records, key=attrgetter('created_at_time'), reverse=True)
+            sorted_records = sorted(records, key=attrgetter('confirmed_at_index'), reverse=True)
```
Comment (paninaro) [inline chia/wallet/trading/trade_store.py:263]: both assignments made to the same variable

File: chia/wallet/trading/trade_store.py (hunk 1 above)
Comment (Quexington) [reply chia/wallet/trading/trade_store.py:251]: Hmm yes I suppose it would be wise to add some tests :)

File: chia/wallet/trading/trade_store.py [hunk 3]
```diff
@@ -246,6 +247,40 @@ async def get_all_trades(self) -> List[TradeRecord]:
...
+            statuses.append("")  # This is a catch all for any statuses we have not explicitly designated
+            for status in statuses:
+                for record in sorted_records:
+                    if status in TradeStatus(record.status).name and record not in records:
+                        records.append(record)
+        else:
+            raise ValueError(f"No known sort {sort_key}")
+
+        # Paginate
+        if start > len(records) - 1:
```
Comment (paninaro) [inline chia/wallet/trading/trade_store.py:279]: Minor nit, and it probably doesn't matter if we're going to be moving to a cursor approach, but this check could be moved to the top, right after the call to `get_all_trades()`

**Review by paninaro — APPROVED** [review]

**Comment by wjblanke** [comment]
wrt/breaking DB change do we need to delete old PCR dbs before running this?

**Pull request merged**